yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -163,13 +180,13 @@ export class PeerManager implements PeerManagerInterface {
    */
   private handleConnectedPeerEvent(e: CustomEvent<PeerId>) {
     const peerId = e.detail;
-    if (this.peerDiscoveryService.isBootstrapPeer(peerId)) {
-      this.logger.verbose(`Connected to bootstrap peer ${peerId.toString()}`);
-    } else {
-      this.logger.verbose(`Connected to transaction peer ${peerId.toString()}`);
+    if (this.config.p2pDisableStatusHandshake) {
+      return;
     }
-    if (!this.config.p2pDisableStatusHandshake) {
+    if (!this.config.p2pAllowOnlyValidators) {
       void this.exchangeStatusHandshake(peerId);
+    } else {
+      void this.exchangeAuthHandshake(peerId);
```

Comment:
Shouldn't we only send the auth handshake if they are not in the trusted peer set? How do supernodes connect between each other otherwise? Or if a peer is trusted, then they do not hit this event?

Reply:
This is probably a good idea. I don't really know what the `trustedPeers` is intended to be used for, but it would probably make sense to not authenticate trusted peers.